app/controllers/users_controller.rb

```diff
@@ -7,25 +7,13 @@ def new
 
   def create
     @user = User.new(user_params)
-    if @user.valid?
-      charge = StripeWrapper::Charge.create(
-        :amount => 999,
-        :card => params[:stripeToken],
-        :description => "Sign up charge for #{@user.email}."
-      )
-      if charge.successful?
-        @user.save
-        handle_invitation
-        AppMailer.send_welcome_email(@user).deliver
-        flash[:success] = "Thank you for registering with MyFlix."
-        session[:user_id] = @user.id
-        redirect_to sign_in_path
-      else
-        flash[:error] = charge.error_message
-        render :new and return
-      end
-    else
-      flash[:error] = "Invalid user information. Please check the errors below."
+    result = UserSignup.new(@user).sign_up(params[:stripeToken], params[:invitation_token])
+
+    if result.successful?
+      flash[:success] = "Thank you for registering with MyFlix."
+      session[:user_id] = @user.id
+      redirect_to sign_in_path
+    else flash[:error] = result.error_message
```

Comment:
This is just a ruby style but, you should not have code next to `else`, also you should use `flash.now` with render.